packages/flutter/test/material/stepper_test.dart

```diff
@@ -953,4 +953,90 @@ void main() {
 
      expect(material.elevation, 2.0);
    });
+
+  testWidgets('Stepper horizontal title [HorizontalTitlePosition.bottom] and line [HorizontalLinePosition.top]', (WidgetTester tester) async {
+    int index = 0;
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Material(
+          child: Stepper(
+            horizontalTitlePosition: HorizontalTitlePosition.bottom,
+            horizontalLinePosition: HorizontalLinePosition.top,
+            onStepTapped: (int i) {
+              index = i;
+            },
+            steps: const <Step>[
+              Step(
+                title: Text('Step 1'),
+                content: SizedBox(
+                  width: 100.0,
+                  height: 100.0,
+                ),
+              ),
+              Step(
+                title: Text('Step 2'),
+                content: SizedBox(
+                  width: 100.0,
+                  height: 100.0,
+                ),
+              ),
+              Step(
+                title: Text('Step 3'),
+                content: SizedBox(
+                  width: 100.0,
+                  height: 100.0,
+                ),
+              ),
+            ],
+          ),
+        ),
+      ),
+    );
+    await tester.tap(find.text('Step 2'));
+    expect(index, 1);
```

Comment:
Verifying the tap effect on these Stepper widgets doesn't add any specific checks for our new rendering logic. At a minimum, I recommend adding `Key` attributes to relevant widgets (the `SizedBox(height: 48)`, `SizedBox(height:0)`, and 1px Container that draws the line are good candidates). This will allow new tests to verify that various parameter pairings have the desired internal effect.